state/protocol/badger/validity_test.go

```diff
@@ -101,3 +115,121 @@ func TestBootstrapInvalidEpochCommit(t *testing.T) {
 		require.Error(t, err)
 	})
 }
+
+func TestValidateVersionBeacon(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("no version beacon is ok", func(t *testing.T) {
+		t.Parallel()
+
+		snap := new(mock.Snapshot)
+
+		vb := &flow.VersionBeacon{}
+
+		vbHeight := uint64(0)
+
+		snap.On("VersionBeacon").Return(vb, vbHeight, state.ErrNoVersionBeacon)
+
+		err := validateVersionBeacon(snap)
+		require.NoError(t, err)
+	})
+
+	t.Run("height must be below highest block", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("height must be below snapshot head", func(t *testing.T) {
```

Reply:
VB event will most of the time be send as a service event included in a block, so I don't think changing it to snapshot head is valid here.

Reply:
The "snapshot head" and "highest block (in sealing segment)" are the same block, so I'm not sure why the change would be invalid. The suggestion is aiming to make the meaning clearer rather than change it. We could also say `below highest block in sealing segment`.

Reply:
Yes, however the validation happens not only in a context of a snapshot but also when event is emitted (or included) so that's why I find this notion to be more universal